.github/scripts/synchronize_jira_xray.js

```diff
@@ -0,0 +1,321 @@
+        },
+      }
+    );
+
+    if (response.status !== 200) {
+      console.log(
+        `Jira API Request Failed with Status Code: ${response.status}`
+      );
+      console.log(response.data);
+      return;
+    }
+
+    console.log(`The ID of the testSet is: ${response.data.id}`);
+    return response.data.id;
+  } catch (error) {
+    console.log(`Error fetching Jira issue: ${error}`);
+  }
+
+  return null;
+}
+
+async function uploadFeatureFileToXray(featureFilePath, XRAY_TOKEN) {
+  try {
+    const formData = new FormData();
+    formData.append("file", fs.createReadStream(featureFilePath));
+
+    const response = await axios.post(XRAY_API_URL, formData, {
+      headers: {
+        ...formData.getHeaders(),
+        Authorization: `Bearer ${XRAY_TOKEN}`,
+      },
+    });
+    if (response.status !== 200) {
+      console.log(
+        `Feature File Upload to Xray Failed with Status Code: ${response.status}`
+      );
+      console.log(response.data);
+      return;
+    }
+    console.log("Feature file uploaded successfully to Xray.");
+    console.log(response.data);
+    return response.data;
+  } catch (error) {
+    console.log(`Error uploading feature file to Xray: ${error}`);
+  }
+
+  return null;
+}
+
+async function updateJiraIssues(testSelfs, targetVersions, componentsList) {
+  for (const api of testSelfs) {
+    try {
+      // Get the existing issue data
+      const response = await axios.get(api, {
+        headers: {
+          Accept: "application/json",
+        },
+        auth: {
+          username: JIRA_USER,
+          password: JIRA_TOKEN_TEST,
+        },
+      });
+
+      if (response.status !== 200) {
+        console.log(
+          `Jira Issue Update Failed with Status Code: ${response.status}`
+        );
+        console.log(response.data);
+        return;
+      }
+
+      const existingIssueData = response.data;
+      const existingCustomField10901 =
+        existingIssueData.fields.customfield_10901 || [];
+      const existingValues = existingCustomField10901
+        ? existingCustomField10901.map((item) => item.value)
+        : [];
+
+      console.log("Existing values of version: ", existingValues);
+
+      for (const targetVersion of targetVersions) {
+        if (!existingValues.includes(targetVersion)) {
+          existingValues.push(targetVersion);
+        }
+      }
+      console.log("the new target versions are: ", existingValues);
+
+      const issueUpdatePayload = {
+        fields: {
+          customfield_10901: existingValues.map((value) => ({
+            value: value,
+          })),
+        },
+      };
+      console.log(
+        "the issue update for",
+        api,
+        " is : ",
+        JSON.stringify(issueUpdatePayload)
+      );
+
+      if (componentsList) {
+        issueUpdatePayload.fields.components = componentsList.map(
+          (component) => ({ name: component })
+        );
+      } else {
+        console.log("No component mentioned");
```

Comment:
should we call api if no component is mentioned ?
Or add `return;` ?

Reply:
@kduret  yes we should, for now we are forcing the component to be `centreon-web` because we are working on the `centreon/centreon` project. Afterward, we will make it more dynamic based on the project.